src/ECL/ECL.hs

```diff
@@ -68,23 +104,30 @@ genericMerge f bsl@((s, b):bs) bsr@((s', b'):bs')
 
 -- | Returns the uncovered set after one binder is applied to the set of
 -- values represented by another.
-missingSingle :: (Eq lit) => Binder lit -> Binder lit -> Binders lit
-missingSingle _ (Var _) = []
-missingSingle b@(Var _) (Tagged tag bc) =
-  map (Tagged tag) $ missingSingle b bc
-missingSingle c@(Tagged tag b) (Tagged tag' b')
-  | tag == tag' = map (Tagged tag) $ missingSingle b b'
-  | otherwise = [c]
-missingSingle (Var _) (Record bs) =
-  map (Record . zip names) miss
+missingSingle :: (Eq lit) => Environment -> Binder lit -> Binder lit -> (Binders lit, Maybe Bool)
+missingSingle _ _ (Var _) = ([], pure True)
+missingSingle env (Var _) cb@(Tagged _ _) =
```

Comment:
This seems to assume that there is only one type of `Tagged` thing, which is fine if you only care about `Nat`s but not if you have 2 or more type constructors.